src/main/resources/avro/reads.avdl

```diff
@@ -177,7 +177,38 @@ record LinearAlignment {
 }
 
 /**
-Each read alignment describes a linear alignment with additional information
+A string-to-reference-graph alignment can be represented by one CIGAR string and
+one `Path` through multiple `Reference`s, against which the CIGAR string is
+interpreted.
+
+Note that `Path`s are restricted to following reference adjacencies.
+If you want to align your read following adjacencies that are
+not present in the `ReferenceSet` to which you are aligning, you have a
+"chimeric" alignment, and should use multiple `ReadAlignment`s and the
+supplementaryAlignment flag instead of a single `GraphAlignment`.
+
+Some especially large deletions could be represented just as well as a large
+deletion in the CIGAR string, or as a chimeric alignment.
+*/
+record GraphAlignment {
```

Comment:
Yeah, you aren't supposed to have a `Segment` on a sequence with no `Reference` in the middle of a `Path` in a `GraphAlignment`. The `ReadGroup` is supposed to be aligned against just a `ReferenceSet`. If we want to support aligning to non-reference sequences, we'll need to add a method in the reads API to let `ReadGroup` carry augmentations for the reference just like `VariantSet` does.